test/e2e/network_policy.go

```diff
@@ -0,0 +1,438 @@
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+/*
+These Network Policy tests create two services A and B.
+There is a single pod in the service A , and a single pod on each node in
+the service B.
+
+Each pod is running a network monitor container (see test/images/network-monitor):
+-  The A pod uses service discovery to locate the B pods and waits
+   to establish communication (if expected) with those pods.
+-  Each B pod uses service discovery to locate the A pod and waits to
+   establish communication (if expected) with that pod.
+
+We test the following:
+-  Policy on, or off.  When policy is on we expect isolation between all pods across the
+   namespaces.
+-  Policy on, but with rules allowing TCP communication between the pods in two services.
+
+TODO:
+-  Test UDP traffic
+*/
+
+// Summary is the data returned by /summary API on the network monitor container
+type Summary struct {
+	TCPNumOutboundConnected int
+	TCPNumInboundConnected  int
+}
+
+const (
+	serviceAName        = "service-a"
+	serviceBName        = "service-b"
+	netMonitorContainer = "gcr.io/google_containers/netmonitor:1.0"
+	convergenceTimeout  = 1 // Connection convergence timeout in minutes
+)
+
+var _ = framework.KubeDescribe("NetworkPolicy", func() {
+	f := framework.NewDefaultFramework("network-policy")
+
+	BeforeEach(func() {
+		// Assert basic external connectivity.
+		// Since this is not really a test of Kubernetes in any way, we
+		// leave it as a pre-test assertion, rather than a Gingko test.
+		By("Executing a successful http request from the external internet")
+		resp, err := http.Get("http://google.com")
+		if err != nil {
+			framework.Failf("Unable to connect/talk to the internet: %v", err)
+		}
+		if resp.StatusCode != http.StatusOK {
+			framework.Failf("Unexpected error code, expected 200, got, %v (%v)", resp.StatusCode, resp)
+		}
+	})
+
+	It("should isolate containers in the same namespace when NetworkIsolation is enabled [Feature:NetworkPolicy]", func() {
+		nsA := f.Namespace
+		runIsolatedToBidirectionalTest(f, nsA, nsA)
+	})
+
+	It("should isolate containers in different namespaces when NetworkIsolation is enabled [Feature:NetworkPolicy]", func() {
+		// This test use two namespaces.  A single namespace is created by
+		// default.  Create another.
+		nsA := f.Namespace
+		nsB, err := f.CreateNamespace(f.BaseName+"-b", map[string]string{
+			"e2e-framework": f.BaseName + "-b",
+		})
+		Expect(err).NotTo(HaveOccurred())
+		runIsolatedToBidirectionalTest(f, nsA, nsB)
+	})
+})
+
+// Run a test that starts with fully isolated pods in two services (in different namespaces) and adds
+// policy objects to allow mono-directional and then bi-directional traffic between the pods in the
+// two services.
+func runIsolatedToBidirectionalTest(f *framework.Framework, nsA, nsB *api.Namespace) {
+
+	var err error
+
+	// Turn on isolation on both namespaces.
+	setNetworkIsolationAnnotations(f, nsA, true)
```

Comment:
Similar to other comments, this should probably be something like `setIngressIsolation(f, nsA, "DefaultDeny")`